app/index.js

```diff
@@ -39,6 +39,29 @@ app.on('window-all-closed', () => {
   }
 });
 
+let quitIsConfirmed = false;
+
+app.on('before-quit', (event) => {
+  const windows = BrowserWindow.getAllWindows().filter(win => win.sessions);
```

Comment:
Smart! I'm currently working on fixing the terminal input issues, and to finish this one i need a way to know if a terminal is currently running something

>The way mac OS Terminal.app handles this is by showing a dialog warning about the currently running process ONLY when trying to quit or close the tab in which a process is running. I'd definitely like to see HyperTerm do the same. No preference needed for that, and no warning when there are no processes running.

will take time to research ways to do that as soon as I'm done with the input fixes :)